File: src/main/java/player/ki/FullMovePruner.java
/*
 * Created on 08.11.2006
 */

package player.ki;

public class FullMovePruner
extends SimplePruner
{
	public FullMovePruner()
	{
		this( 1,2,2 );
	}
	
	public FullMovePruner( int no_hit_radius,int figure_hit_radius,int chess_radius )
	{
		super( no_hit_radius,figure_hit_radius,chess_radius );
	}
	
	public boolean stopMinDynamicEvaluation( int depth,
	                                         boolean thisMove_isChess,
	                                         boolean thisMove_hasHitFigure,
	                                         boolean lastMove_isChess,
	                                         boolean lastMove_hasHitFigure )
	{		
		return false;
	}
}

File: src/main/kotlin/voidchess/player/ki/FullMovePruner.kt
/*
 * Created on 08.11.2006
 */

package voidchess.player.ki

class FullMovePruner constructor(no_hit_radius: Int, figure_hit_radius: Int, chess_radius: Int): SimplePruner(no_hit_radius, figure_hit_radius, chess_radius) {

    override fun stopMinDynamicEvaluation(depth: Int,
                                          thisMove_isChess: Boolean,
                                          thisMove_hasHitFigure: Boolean,
                                          lastMove_isChess: Boolean,
                                          lastMove_hasHitFigure: Boolean): Boolean {
        return false
    }
}
